Automattic/wp-calypso
# Support new Intro Offers Endpoint

#### Changes proposed in this Pull Request

- Support new `/introductory-offers` endpoint ( D72857-code )
- Use introductory offers to display discount prices on the variant picker in the cart

| Before ( introductory offer not displayed on variant ) | After ( introductory offer displayed on variant ) |
| ------------------------------------------------------ | --------------------------------------------------------|
| <img width="573" src="https://user-images.githubusercontent.com/2810519/149253615-28c33e3b-c1fc-40af-b582-b0770da9d20c.png"> | <img width="573" src="https://user-images.githubusercontent.com/2810519/149253616-3ed61def-1007-4719-9eb3-2a1f57cb2aec.png"> |



#### Testing instructions



1. Add an assortment of Jetpack annual plans and products to the cart
2. Click "Edit" in cart to open the product variant selector
2. Verify that in the product variant picker the correct introductory offer ( 50% off ).
3. Verify the cart total is correct.
4. Add an annual product to the cart via `cloud.jetpack.com/pricing` in a logged-out/incognito window.
5. Verify the correct discount and price is shown on the variant selector.



Related to 1201295898168937-as-1201646705905156
